### DISCUSSION TITLE
How to build a "release" Docker image

### DISCUSSION THREAD

**Discussion by Rolf-Smit** [photoprism/photoprism#2282] [Q&A]
I would like to make some local changes to PhotoPrism, then publish a local Docker image that basically mimics a normal release as much as possible. The PhotoPrism repository contains quite some Docker files, and I'm a bit lost which is which.

I noticed there is a `photoprism/docker/develop` folder with base Docker files (images): https://github.com/photoprism/photoprism/tree/develop/docker/develop/bookworm

1. Are these "develop" Docker files also used as base for a release image (they seem to be)?
2. Are the Docker files within `photoprism/docker/photoprism` meant to build release like images? I see they use a multi-stage pattern so I assume they are?

**Comment by ttmms** [comment]
Did you figure it out? Basically I want to build my own fork, but the development guide is very convoluted. And for example `make docker-build` step just hangs indefinitely without any output.

Makefile seems to have some recursion problem, running make with debug info, I just get infinite spam about 

```
...

Makefile:25: not recursively expanding INSTALL_PATH to export to shell function
Makefile:21: not recursively expanding BUILD_ARCH to export to shell function
Makefile:22: not recursively expanding JS_BUILD_PATH to export to shell function
Makefile:25: not recursively expanding INSTALL_PATH to export to shell function

...
```
EDIT: Above is caused by make 4.4 update. Fixed by replacing `?=`s with `:=`s in Makefile for variables that use shell commands

Now building as in manual works, but still quite annoying that it's so complicated. I want to replace URL to point to my fork and have everything else as usual

**Reply by thoroc** [reply]
That piece of information on Make 4.4. saved my bacon in a completely unrelated project.

**Comment by xiaobai6769** [comment]
Is there any way to build an image?

**Reply by lastzero** [reply]
You can, for example, run `make docker-local` in a terminal:
https://github.com/photoprism/photoprism/blob/546635f8b6b4ad906649a0d3daccc30200b98f4f/Makefile#L746
